src/main/mkdir.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2015 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import fs from 'fs-extra';
```

Comment:
I have been thinking, can we just use the built-in `fs/promises` in node instead of this library? Seems to have most of the functions we need.

Reply:
Yes, good idea. No idea how much effort that is. I will have a look.

Reply:
I had a look at it but I don't want to do it now. 

In some files we have a mixture of sync and async uses of `fs-extra`. In some places we use functions added by `fs-extra`, such as [`remove`](https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md). It didn't feel safe to change all of these without checking that it doesn't break anything, so I left things as they are. 

But I agree that we should push towards only using node’s `fs` when we touch things.